block-producer/src/server/api.rs

```diff
@@ -21,38 +32,138 @@ use crate::{
     SharedProvenTx,
 };
 
-struct RpcStore {}
+struct DefaultStore {
+    store: store_client::ApiClient<Channel>,
+}
 
 #[async_trait]
-impl ApplyBlock for RpcStore {
+impl ApplyBlock for DefaultStore {
     async fn apply_block(
         &self,
-        _block: Arc<Block>,
+        block: Arc<Block>,
     ) -> Result<(), ApplyBlockError> {
-        todo!()
+        let request = tonic::Request::new(ApplyBlockRequest {
+            block: Some(block.header.into()),
+            accounts: block
+                .updated_accounts
+                .iter()
+                .map(|(account_id, account_hash)| AccountUpdate {
+                    account_id: Some((*account_id).into()),
+                    account_hash: Some(account_hash.into()),
+                })
```

Comment:
suggestions:

- add `impl From<Account> for AccountUpdate` as a [conversion](https://github.com/0xPolygonMiden/miden-node/blob/main/proto/src/conversion.rs).
- use [`convert`](https://github.com/0xPolygonMiden/miden-node/blob/main/proto/src/conversion.rs#L332-L338)

should make the code a bit cleaner

Reply:
changed all of them to use `convert`, and created #99 to address the `clone`s